mava/wrappers/jaxmarl.py

```diff
@@ -297,23 +281,55 @@ def discount_spec(self) -> specs.BoundedArray:
 
     def action_mask(self, state: JaxMarlState) -> Array:
         """Get action mask for each agent."""
-        if self.has_action_mask:
-            avail_actions = self._env.get_avail_actions(state)
-            mask = jnp.array(batchify(avail_actions, self.agents), dtype=jnp.float32)
-        else:
-            mask = jnp.ones(self._action_shape, dtype=jnp.float32)
-        return mask
+        pass
+
+    def get_global_state(self, env_state: Array, obs: Dict[str, Array]) -> Array:
+        """Get global state from observation for each agent."""
+        pass
+
 
-    def get_global_state(self, state: Array, obs: Dict[str, Array], agents_view: Array) -> Array:
+class SmaxWrapper(JaxMarlWrapper):
+    """WWrapper for SMAX environment"""
+
+    def __init__(self, env: MultiAgentEnv, has_global_state: bool = False, timelimit: int = 500):
+        state_size = env.state_size
+        super().__init__(env, has_global_state, state_size, timelimit)
+
+    def action_mask(self, state: JaxMarlState) -> Array:
+        """Get action mask for each agent."""
+        avail_actions = self._env.get_avail_actions(state)
+        return jnp.array(batchify(avail_actions, self.agents), dtype=jnp.float32)
+
+    def get_global_state(self, env_state: Array, obs: Dict[str, Array]) -> Array:
         """Get global state from observation and copy it for each agent."""
-        if hasattr(obs, "world_state"):
-            global_state = jnp.tile(jnp.array(obs["world_state"]), (self._env.num_agents, 1))
-        elif isinstance(self._env, MABraxEnv):
-            # Use the global state of brax.
-            global_state = jnp.tile(state.obs, (self._env.num_agents, 1))
+        return jnp.tile(jnp.array(obs["world_state"]), (self.num_agents, 1))
 
-        else:
-            global_state = jnp.concatenate(agents_view, axis=0)
-            global_state = jnp.tile(global_state, (self._env.num_agents, 1))
+
+class MabraxWrapper(JaxMarlWrapper):
+    """Wrraper for the Mabrax environment."""
+
+    def __init__(self, env: MABraxEnv, has_global_state: bool = False, timelimit: int = 1000):
+
+        state_size = env.env.observation_size
+        state_size = (
+            state_size if env.homogenisation_method != "max" else state_size + env.num_agents
+        )
+        super().__init__(env, has_global_state, state_size, timelimit)
+        self.action_shape = self.action_spec().shape
+
+    def action_mask(self, state: JaxMarlState) -> Array:
+        """Get action mask for each agent."""
+        return jnp.ones(self.action_shape, dtype=jnp.float32)
+
+    def get_global_state(self, env_state: Array, obs: Dict[str, Array]) -> Array:
+        """Get global state from observation and copy it for each agent."""
+        # Use the global state of brax.
+        global_state = jnp.tile(env_state.obs, (self.num_agents, 1))
```

Comment:
It would be great to see if we can get the underlying envs obs to put here, I'll have a quick look into this :thinking: